lib/adding_columns.py

```diff
@@ -91,13 +98,23 @@ def add_average_films_ratings(spark: SparkSession, data: DataFrame) -> DataFrame
         .select(["nconst", "average_films_rating"])
     )
     data = data.join(data_with_ratings, on="nconst", how="left")
+    data = data.na.fill(value = 0, subset = ["average_films_rating"])
     return data
 
+def add_normalized_columns(data: DataFrame) -> DataFrame:
+    unlist = udf(lambda x: round(float(list(x)[0]),3), DoubleType())
+    for i in ["no_nominations_oscars", "no_oscars", "no_nominations_globes", "no_globes", "no_nominations_emmy", "no_emmy", "no_films", "average_films_rating"]:
+        assembler = VectorAssembler(inputCols=[i],outputCol=i+"_Vect")
+        scaler = MinMaxScaler(inputCol=i+"_Vect", outputCol=i+"_norm")
+        pipeline = Pipeline(stages=[assembler, scaler])
```

Comment:
może wyzwanie - ale jeśli nie pójdzie łatwo to zostaw jako zadanie. 
Operacje withcolumn czy drop też powinno się dać wstawić jako element pipeline.
Tu podpowiedź jak to można zrobić [https://towardsdatascience.com/pyspark-wrap-your-feature-engineering-in-a-pipeline-ee63bdb913](https://towardsdatascience.com/pyspark-wrap-your-feature-engineering-in-a-pipeline-ee63bdb913)

Reply:
zrobione